assets/src/blocks/Columns/ColumnsEditor.js

```diff
@@ -0,0 +1,185 @@
+                <URLInput
+                  label={__('Column %s link', 'planet4-blocks-backend').replace('%s', index + 1)}
+                  placeholder={__('Enter link for column %s', 'planet4-blocks-backend').replace('%s', index + 1)}
+                  value={cta_link}
+                  onChange={toAttribute('cta_link', index)}
+                />
+                <CheckboxControl
+                  label={__('Open link in new tab', 'planet4-blocks-backend')}
+                  value={link_new_tab}
+                  checked={link_new_tab}
+                  onChange={toAttribute('link_new_tab', index)}
+                />
+              </div>
+            );
+          })}
+        </PanelBody>
+      </InspectorControls>
+    </>
+  );
+};
+
+export const ColumnsEditor = ({ isSelected, attributes, setAttributes }) => {
+  const {
+    columns,
+    className,
+    columns_description,
+    columns_title,
+    columns_block_style,
+    isExample,
+    exampleColumns,
+  } = attributes;
+
+  // Update column block style based on className attribute
+  useEffect(() => {
+    if (className && className.includes('is-style-')) {
+      const newColumnsStyle = className.split('is-style-')[1];
+      setAttributes({
+        columns_block_style: newColumnsStyle
+      });
+    }
+  }, [className]);
+
+  const { postType } = useSelect(select => ({
+    postType: select('core/editor').getCurrentPostType()
+  }), []);
+
+  const { columnImages } = useSelect(select => {
+    let columnImages = [];
+    columns.forEach(column => {
+      if (column.attachment && column.attachment > 0) {
+        const media_details = select('core').getMedia(column.attachment);
+        if (media_details) {
+          columnImages[column.attachment] = select('core').getMedia(column.attachment).source_url;
+        }
+      }
+    });
+    return { columnImages };
+  }, [columns]);
+
+  const toAttribute = (attributeName, index) => value => {
+    if (['columns_title', 'columns_description'].includes(attributeName)) {
+      setAttributes({
+        [attributeName]: value
+      });
+    } else {
+      let columns = JSON.parse(JSON.stringify(attributes.columns));
+      columns[index][attributeName] = value;
+      setAttributes({ columns });
+    }
+  }
+
+  return (
+    <section className={`block columns-block block-style-${columns_block_style}`}>
+      <div className='container'>
+        {renderEdit(attributes, toAttribute, setAttributes, isSelected)}
+        {!isExample &&
+          <header className='articles-title-container'>
+            <RichText
+              tagName='h2'
+              className='page-section-header'
+              placeholder={__('Enter title', 'planet4-blocks-backend')}
+              value={columns_title}
+              onChange={toAttribute('columns_title')}
+              keepPlaceholderOnFocus={true}
+              withoutInteractiveFormatting
+              allowedFormats={[]}
+              characterLimit={40}
```

Comment:
I realized this week (well React insulted me) that `characterLimit` is not a RichText property (and [doesn't look like](https://github.com/WordPress/gutenberg/issues/22972) it's gonna be) but [our own thing](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/blob/master/assets/src/components/withCharacterCounter/withCharacterCounter.js). Not sure what to do with this information.

Reply:
Yes I've noticed that too 😁 not sure what to do about it either... It's in so many of our blocks at the moment! From a design perspective it's tempting to add character limits, but we support so many languages that I'm not sure it makes sense in our case, maybe we should just remove this all together.